File: solutions/3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.cs
public class Solution {
    public string ShortestBeautifulSubstring(string s, int k) {
        for (int m = k; m <= s.Length; m++) {
            string ans = "";
            for (int i = m; i <= s.Length; i++) {
                string t = s.Substring(i - m, m);
                if ((ans.Length == 0 || string.CompareOrdinal(t, ans) < 0) &&
                    t.Count(c => c == '1') == k) {
                    ans = t;
                }
            }
            if (ans.Length > 0) {
                return ans;
            }
        }
        return "";
    }
}

File: solutions/3150-shortest-and-lexicographically-smallest-beautiful-string/shortest-and-lexicographically-smallest-beautiful-string.go
func shortestBeautifulSubstring(s string, k int) string {
	for m := k; m <= len(s); m++ {
		ans := ""
		for i := m; i <= len(s); i++ {
			t := s[i-m : i]
			if (ans == "" || t < ans) && strings.Count(t, "1") == k {
				ans = t
			}
		}
		if ans != "" {
			return ans
		}
	}
	return ""
}